sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3635,6 +3637,8 @@ object functions {
    * @param e a column containing a struct, an array or a map.
    * @param options options to control how the struct column is converted into a json string.
    *                accepts the same options and the json data source.
+   *                Additionally the function supports the `pretty` option which enables
```

Comment:
For clarification, we don't support this in JSON datasource officially since it's not documented.